cmd/katalyst-agent/app/options/sysadvisor/qosaware/resource/memory/headroom/policy_canonical.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2022 The Katalyst Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package headroom
+
+import (
+	"github.com/spf13/pflag"
+
+	"github.com/kubewharf/katalyst-core/pkg/config/agent/sysadvisor/qosaware/resource/memory/headroom"
+)
+
+const (
+	defaultEnableBuffer                    = false
+	defaultMemoryFreeEstimationUtilization = 0.6
+	defaultStaticMemoryOversold            = 20 << 30 // 20GB
+
+	defaultCacheOversoldRate                     = 0
+	defaultCacheOversoldMemoryCPURatioUpperBound = 6.0
+	defaultCacheOversoldMemoryCPURatioLowerBound = 3.5
+)
+
+type MemoryPolicyCanonicalOptions struct {
+	EnableBuffer                          bool
+	MemoryFreeEstimationUtilization       float64
```

Comment:
// add comments here to explain we will choose the max of them
type MemoryUtilBasedOptions struct {
   FreeBasedRatio
   CacheBasedRatio
   StaticBasedCapacity
   
   Enabled
   CPUMemRatioLowerBound
   CPUMemRatioUpperBound
}